skyportal/handlers/api/observation_plan.py

```diff
@@ -207,40 +205,60 @@ def post_survey_efficiency_analysis(
         f'Simsurvey analysis in progress for ID {survey_efficiency_analysis.id}. Should be available soon.'
     )
 
-    simsurvey_analysis = functools.partial(
-        observation_simsurvey,
-        observations,
-        localization.id,
-        instrument.id,
-        survey_efficiency_analysis.id,
-        "SurveyEfficiencyForObservationPlan",
-        width=width,
-        height=height,
-        number_of_injections=payload['numberInjections'],
-        number_of_detections=payload['numberDetections'],
-        detection_threshold=payload['detectionThreshold'],
-        minimum_phase=payload['minimumPhase'],
-        maximum_phase=payload['maximumPhase'],
-        model_name=payload['modelName'],
-        optional_injection_parameters=payload['optionalInjectionParameters'],
-    )
+    if asynchronous:
+        simsurvey_analysis = functools.partial(
+            observation_simsurvey,
+            observations,
+            localization.id,
+            instrument.id,
+            survey_efficiency_analysis.id,
+            "SurveyEfficiencyForObservationPlan",
+            width=width,
+            height=height,
+            number_of_injections=payload['numberInjections'],
+            number_of_detections=payload['numberDetections'],
+            detection_threshold=payload['detectionThreshold'],
+            minimum_phase=payload['minimumPhase'],
+            maximum_phase=payload['maximumPhase'],
+            model_name=payload['modelName'],
+            optional_injection_parameters=payload['optionalInjectionParameters'],
+        )
 
-    IOLoop.current().run_in_executor(None, simsurvey_analysis)
+        IOLoop.current().run_in_executor(None, simsurvey_analysis)
+    else:
+        observation_simsurvey(
+            observations,
+            localization.id,
+            instrument.id,
+            survey_efficiency_analysis.id,
+            "SurveyEfficiencyForObservationPlan",
+            width=width,
+            height=height,
+            number_of_injections=payload['numberInjections'],
+            number_of_detections=payload['numberDetections'],
+            detection_threshold=payload['detectionThreshold'],
+            minimum_phase=payload['minimumPhase'],
+            maximum_phase=payload['maximumPhase'],
+            model_name=payload['modelName'],
+            optional_injection_parameters=payload['optionalInjectionParameters'],
+        )
 
     return survey_efficiency_analysis.id
 
 
-def post_observation_plans(plans, user_id, session):
+def post_observation_plans(plans, user_id, session, asynchronous=True):
```

Comment:
In this method, you do not use the `asynchronous` parameter. You only seem to use it in the `post_observation_plans` one (singular, without the s). Please consider either using it or removing it.

Reply:
Fixed.